build.gradle

```diff
@@ -53,26 +56,61 @@ dependencies {
     testImplementation 'junit:junit:4.12'
 }
 
-// create a release zip
-task release(type: Zip) {
+def readModInfoJson() {
     def jsonFile = file('src/main/resources/mod_info.json')
-    def slurper = new groovy.json.JsonSlurper()
-    def parsedJson = slurper.parse(jsonFile)
+    def slurper = new JsonSlurper()
+    return slurper.parse(jsonFile)
+}
+
+task release(type: Zip) {
     dependsOn build
-    archiveFileName = "${project.name}-${parsedJson.version}.zip"
+    archiveFileName = "${project.name}-${readModInfoJson().version}.zip"
     destinationDirectory = file(rootDir)
     from sourceSets.main.resources
     from jar
     into project.name
 }
 
+def arePropertiesDefined() {
+    if (!project.hasProperty('game.dir') || !project.hasProperty('game.mod_folder')) {
+        println "WARNING: Missing properties in gradle.properties. See gradle.properties.dist as an example."
+        println "WARNING: Task execution stopped."
+        return false;
+    }
+    return true;
+}
+
+def getModDir() {
+    return project.getProperty('game.dir') + project.getProperty('game.mod_folder') + readModInfoJson().id
+}
+
 task copyToMods(type: Copy) {
-    if (!project.hasProperty('game.full_path')) {
-        println "WARNING: Missing game.full_path in gradle.properties. See gradle.properties.dist as an example."
+    if (!arePropertiesDefined()) {
         return;
     }
 
     dependsOn build
     from jar
-    into project.getProperty('game.full_path')
-}
\ No newline at end of file
+
+    into getModDir()
+}
+
+def getCoreDir() {
+    return project.getProperty('game.dir') + "/starsector-core/"
+}
+
+task runGameWithBuild(type:Exec) {
+    if (!arePropertiesDefined()) {
+        return;
+    }
+
+    dependsOn copyToMods
+
+    workingDir = file(getCoreDir())
+
+    if (Os.isFamily(Os.FAMILY_WINDOWS)) {
+        commandLine 'cmd', '/c', 'starsector.bat'
```

Comment:
Can we make it another property defined by the user? That way we can avoid os check and just execute whatever user specifies in a folder that user specified as well.

Reply:
What's wrong with OS check?

Reply:
Was hoping for a better solution regarding portability. if-else always feels like a hack.